### PR TITLE
Fix crash in l_body_get_ground_position if there is no body to get po…

### PR DESCRIPTION
**Pull request by JonBooth78**
…sition relative to

Every time I try to hyperspace I was crashing here (jumping from a space station, if that matters).  Well the crash happened in `GetAltitudeRelTo` which was being passed a `null` body as the first argument.

@Max5377 , I think this relates to your earlier PR.

### CODE REVIEW

**Comment by Max5377** [comment]
I haven't considered that this can happen when doing `PR`. Thanks for spotting this.
P.S.: Maybe the better way would be to return `null` in `Frame::GetFrame` if no `body` is in this frame?

**Comment by JonBooth78** [comment]
Hey @Max5377 , thanks for looking at this.

I *think* a frame reference with no body attached is fine and so we probably should still return that frame.

I'm not really across all the details here so if you'd like to progress this, how about you create a PR from your branch and then I'll close this one?

**Comment by Max5377** [comment]
@JonBooth78 `cherry-pick` is a command that basically allows you to copy someone else's commit into your branch. https://docs.github.com/en/desktop/managing-commits/cherry-picking-a-commit-in-github-desktop
https://git-scm.com/docs/git-cherry-pick
https://www.atlassian.com/git/tutorials/cherry-pick

You can also give write access to your repo by doing this
https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-teams-and-people-with-access-to-your-repository
so I can quickly change your commit.
I would modify this PR directly, but only mainteiners can do that.

I could also do PR to your branch, but I don't know if this would cause merge conflicts, because from the looks you haven't synced your fork in a long time.
I don't really want to do additional PR since, you've already done one that addresses this.

**Comment by JonBooth78** [comment]
Hey @Max5377,I know how to pull more of your changes into this PR and I've done it for now.  I'm just a bit concerned if there are more changes at code review, I might slow things down :-)

If you wanted to do a PR into my repo, you wouldn't want to do it to my `master` as yeah, that's a bit behind (and ahead), instead you'd want to do a PR to the source branch of this PR.  Still that would leave me possibly in the middle.  We'll see how we go.

File: src/lua/LuaBody.cpp [hunk 1]
```diff
@@ -668,12 +668,18 @@ static int l_body_get_ground_position(lua_State *l)
 		lua_pushnil(l);
 		return 1;
 	}
+	Body* astro = f->GetBody(); // null
+	if (!astro)
+	{
+		lua_pushnil(l);
+		return 1;
+	}
```
Comment (Web-eWorks) [inline src/lua/LuaBody.cpp:676]: As a nitpick for both @JonBooth78 and @Max5377 - Pioneer's codebase prefers the use of starting braces on the same line for `if` and `for` statements, as well as class/struct definitions and single-line function definitions. Multi-line function definitions should (assuming I haven't forgotten anything) be the only structure with their beginning brace on a newline.

The clang-format CI check would be complaining at you over this... if it hadn't been silently broken for quite some time and I didn't notice. :neutral_face:

**Comment by JonBooth78** [comment]
Ahh, man sorry @Web-eWorks , tidied those up for you now :-)

I'll also update my VS pro settings to remind me!

**Review by Web-eWorks — APPROVED** [review]

**Pull request merged**